### ISSUE TITLE
#[deny(overflowing_literals)] is not triggered on proc macro output without a source span

### ISSUE THREAD

**Issue by kupiakos** [rust-lang/rust#100695]
Proc macros will not trigger `#[deny(overflowing_literals)]` if the violating output literal is not associated with an input span. This is unexpected, as an overflowing literal should always be checked regardless of what generated it.

proc macro lib.rs:
```rust
extern crate proc_macro;

#[proc_macro]
pub fn no_overflow_error_without_span(input: proc_macro::TokenStream) -> proc_macro::TokenStream {
    // Convert the input to/from string to remove span info
    let value_str = input.to_string();
    let mut output = proc_macro::TokenStream::new();
    output.extend([
        "pub const PEAR: u8 = "
            .parse::<proc_macro::TokenStream>()
            .unwrap(),
        value_str.parse().unwrap(),
        ";".parse().unwrap(),
    ]);
    output
}

#[proc_macro]
pub fn overflow_error_with_unrelated_span(input: proc_macro::TokenStream) -> proc_macro::TokenStream {
    let mut input = input.into_iter();
    let in_token = input.next().expect("single token");
    assert!(input.next().is_none(), "more than one token");

    let value_str = "256".to_string();
    let mut value_token: proc_macro::TokenTree = value_str
        .parse::<proc_macro::TokenStream>()
        .unwrap()
        .into_iter()
        .next()
        .unwrap();
    value_token.set_span(in_token.span());
    let mut output = proc_macro::TokenStream::new();
    output.extend([
        "pub const BANANA: u8 = "
            .parse::<proc_macro::TokenStream>()
            .unwrap(),
        value_token.into(),
        ";".parse().unwrap(),
    ]);
    output
}

#[proc_macro]
pub fn overflow_error_with_passthrough(input: proc_macro::TokenStream) -> proc_macro::TokenStream {
    let mut output = proc_macro::TokenStream::new();
    output.extend([
        "pub const APPLE: u8 = "
            .parse::<proc_macro::TokenStream>()
            .unwrap(),
        input,
        ";".parse().unwrap(),
    ]);
    output
}

```

test:
```rust

// Errors out as expected
overflow_error_with_passthrough!(256);

// Gives an expected overflow error even though the span is completely unrelated to the overflow.
overflow_error_with_unrelated_span!(blah);

// No error here! This is unexpected.
no_overflow_error_without_span!(256);
```

`rustc --version --verbose`:
```
$ rustc --version --verbose
rustc 1.64.0-nightly (87588a2af 2022-07-13)
binary: rustc
commit-hash: 87588a2afd9ca903366f0deaf84d805f34469384
commit-date: 2022-07-13
host: x86_64-unknown-linux-gnu
release: 1.64.0-nightly
LLVM version: 14.0.6
```
